### PR TITLE
Warning Cleanup and Clang Compile Fix

### PR DESCRIPTION
**Pull request by adamemerson**
Clean up existing warnings for Clang and GCC

Also fix a compile error in clang.

### CODE REVIEW

**Review by ronen-fr — APPROVED** [review]

File: src/test/librados_test_stub/LibradosTestStub.cc [hunk 1]
```diff
@@ -1455,7 +1455,7 @@ int cls_cxx_list_watchers(cls_method_context_t hctx,
     watcher.name = entity_name_t::CLIENT(w.watcher_id);
     watcher.cookie = w.cookie;
     watcher.timeout_seconds = w.timeout_seconds;
-    watcher.addr.parse(w.addr, 0);
+    watcher.addr.parse(std::string_view(w.addr), 0);
```
Comment (tchaikov) [inline src/test/librados_test_stub/LibradosTestStub.cc:1458]: i'd suggest just drop `0`, like:
```c++
watcher.addr.parse(w.addr);
```

**Review by tchaikov — APPROVED** [review]
LGTM apart from a nit, would be better if it's fixed. moreover, to use `0` for nullptr is not very readable.

File: src/test/librados_test_stub/LibradosTestStub.cc (hunk 1 above)
Comment (adamemerson) [reply src/test/librados_test_stub/LibradosTestStub.cc]: Done!

**Review by wjwithagen — APPROVED** [review]
FreeBSD Clang is happier

**Pull request merged**